### PR TITLE
new WORKING software list additions

### PR DESCRIPTION
**Pull request by Sterophonick**
new WORKING software list additions
channelf: Pac-Man (PD)
mz700_cass: Building Hopper

new NOT_WORKING software list additions
gameboy: Carazu (World, v1.0)

### CODE REVIEW

**Comment by Sterophonick** [comment]
I can't figure out why Carazu (in gameboy.xml) doesn't work? I tried most of the MBCs and it still broke on the second level.

**Comment by DavidHaywood** [comment]
does it work on real hardware?

**Comment by Sterophonick** [comment]
I do not have the means of testing GB / GBC games on real hardware. What I can say is that it does work when loading the rom without the use of the software lists.

**Comment by DavidHaywood** [comment]
ok, then yes, it should work and you're adding the incorrect cart details.

just check what the hardware guess code is guessing it as and use that I guess.

**Comment by cuavas** [comment]
Where did these dumps actually come from?

**Comment by Sterophonick** [comment]
Pac-Man and Carazu actually came from various homebrew sites, and Building Hopper came from a site called RetroPC.net.

**Comment by cuavas** [comment]
Are the homebrew titles at least three years old or released freely by the authors?

**Comment by Sterophonick** [comment]
Yes.

**Comment by Sterophonick** [comment]
Yes.

**Comment by Sterophonick** [comment]
Oh, carazu also doesn't save.

**Comment by DavidHaywood** [comment]
then it probably also needs the correct cart type specifying and the correct sram region specifying.

the thing with the software lists is that they intentionally *won't* resort to the inaccurate guessing, it's a database of what's in the carts, so you need to specify those things.

**Comment by Sterophonick** [comment]
It all works now.

**Comment by Sterophonick** [comment]
@cuavas I fixed the refresh rate.

File: hash/mz700_cass.xml [hunk 1]
```diff
@@ -2842,6 +2842,17 @@ BASIC 5Z-008 V1.0B
 			</dataarea>
 		</part>
 	</software>
+	
+	<software name="bhopper">
+		<description>Building Hopper</description>
+		<year>1988</year>
+		<publisher>Taku Ito.</publisher>
```
Comment (cuavas) [inline hash/mz700_cass.xml]: Why the trailing `.` after the person's name?

File: src/mame/drivers/gigatron.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ void gigatron_state::gigatron(machine_config &config)
 
 	/* video hardware */
 	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
-	screen.set_refresh_hz(59.98);
+	screen.set_refresh_hz(59.94);
```
Comment (cuavas) [inline src/mame/drivers/gigatron.cpp]: I don’t consider this “fixed” if you haven’t actually derived screen timings from the dot clock etc.  Also, unrelated changes shouldn't be mixed up like this.

File: src/mame/drivers/spg2xx_vii.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // license:BSD-3-Clause
 // copyright-holders:Ryan Holtz, David Haywood
 
-/* 'Zone' systems */
+/* Siatronics/KenSingTon/JungleSoft Vii */
```
Comment (cuavas) [inline src/mame/drivers/spg2xx_vii.cpp]: Please don’t make random changes like this.  You’re likely to cause a conflict considering this file is being touched by @DavidHaywood pretty frequently.

File: hash/mz700_cass.xml (hunk 1 above)
Comment (Sterophonick) [reply hash/mz700_cass.xml]: It is stated like that in the game itself.

File: src/mame/drivers/spg2xx_vii.cpp (hunk 3 above)
Comment (Sterophonick) [reply src/mame/drivers/spg2xx_vii.cpp]: Reverted.

File: src/mame/drivers/gigatron.cpp (hunk 2 above)
Comment (Sterophonick) [reply src/mame/drivers/gigatron.cpp]: https://gigatron.io/?page_id=482

"Video: 59.98 Hz VGA compatible (standard DE-15 connector)"

Comment (cuavas) [reply src/mame/drivers/gigatron.cpp]: So is it actually 59.98 or 59.94?  I haven’t looked at the schematics myself, but I’d bet it isn’t actually using the same crystal and dividers as a real VGA card, so it would have different timing parameters.  You need to actually identify the timing circuitry and derive the video parameters from that.

Consider this source file: https://github.com/mamedev/mame/blob/mame0217/src/mame/drivers/osborne1.cpp

There are three machines, each with different screen timing parameters.  Two of them (`osborne1` and `osborne1sp`) derive the dot clock and raster timings from the master clock that’s also divided down for CPU and RAM timing.  The third one (`osborne1nv`) uses a programmable CRT controller with its own crystal.  Note that in all three cases, the screen parameters can be set in terms of the dot clock, using the `set_raw` configuration function.

You have schematics for this machine.  Use them.

**Comment by Sterophonick** [comment]
It is 59.98 Hz.

**Comment by Sterophonick** [comment]
Any updates on this getting merged, or is gbad not necessary?
